src/Sparrow.Server/Utils/DiskStatsGetter/WindowsDiskStatsGetter.cs

```diff
@@ -117,7 +117,18 @@ public DiskCounters Get(string path)
                     if (Logger.IsOperationsEnabled)
                         Logger.Operations($"{nameof(DiskCounters)} was created for \"{drive}\" (requested for path \"{path}\").");
 
-                    counter = _countersPerDisk[path] = new DiskCounters(name);
+                    try
+                    {
+                        counter = _countersPerDisk[path] = new Lazy<DiskCounters>(() => new DiskCounters(name));
+                    }
+                    catch (UnauthorizedAccessException)
+                    {
+                        if (Logger.IsOperationsEnabled)
```

Comment:
is this correct? You are not materializing this here now with Lazy, but few lines below